newIDE/app/src/GameDashboard/LeaderboardAdmin/LeaderboardSortOptionsDialog.js

```diff
@@ -0,0 +1,219 @@
+
+  const [
+    extremeAllowedScoreError,
+    setExtremeAllowedScoreError,
+  ] = React.useState<?string>(null);
+  const [
+    displayExtremeAllowedScoreInput,
+    setDisplayExtremeAllowedScoreInput,
+  ] = React.useState<boolean>(extremeAllowedScore !== undefined);
+  const [
+    extremeAllowedScoreValue,
+    setExtremeAllowedScoreValue,
+  ] = React.useState<number>(extremeAllowedScore || 0);
+
+  const [sortOrder, setSortOrder] = React.useState<LeaderboardSortOption>(
+    sort || 'ASC'
+  );
+
+  const onSaveSettings = async (i18n: I18nType) => {
+    if (displayExtremeAllowedScoreInput) {
+      if (extremeAllowedScoreValue > extremeAllowedScoreMax) {
+        setExtremeAllowedScoreError(
+          i18n._(t`Extreme score must be lower than ${extremeAllowedScoreMax}.`)
+        );
+        return;
+      }
+      if (extremeAllowedScoreValue < extremeAllowedScoreMin) {
+        setExtremeAllowedScoreError(
+          i18n._(
+            t`Extreme score must be equal or higher than ${extremeAllowedScoreMin}.`
+          )
+        );
+        return;
+      }
+    }
+    if (isNaN(extremeAllowedScoreValue)) {
+      setExtremeAllowedScoreError(
+        i18n._(
+          t`Limit cannot be empty, uncheck or fill a value between ${extremeAllowedScoreMin} and ${extremeAllowedScoreMax}.`
+        )
+      );
+      return;
+    }
+    setIsLoading(true);
+    const sortOrderSettings = {
+      sort: sortOrder,
+      extremeAllowedScore: displayExtremeAllowedScoreInput
+        ? extremeAllowedScoreValue
+        : null,
+    };
+    await onSave(sortOrderSettings);
+  };
+
+  return (
+    <I18n>
+      {({ i18n }) => (
+        <Dialog
+          open={open}
+          maxWidth="sm"
+          onRequestClose={() => {
+            if (!isLoading) onClose();
+          }}
+          secondaryActions={[
+            <HelpButton
+              key="help"
+              helpPagePath="/interface/games-dashboard/leaderboard-administration"
+              anchor="change_the_appearance_of_a_leaderboard"
+            />,
+          ]}
+          actions={[
+            <FlatButton
+              label={<Trans>Cancel</Trans>}
+              disabled={isLoading}
+              onClick={onClose}
+              key={'cancel'}
+            />,
+            <RaisedButton
+              primary
+              label={<Trans>Save</Trans>}
+              disabled={isLoading}
+              onClick={() => onSaveSettings(i18n)}
+              key={'save'}
+            />,
+          ]}
+          onApply={() => {
+            onSaveSettings(i18n);
+          }}
+        >
+          <Text size="title">
+            <Trans>Sort order</Trans>
+          </Text>
+          <ColumnStackLayout>
+            <SelectField
+              fullWidth
+              value={sortOrder}
+              floatingLabelText={<Trans>Scores sort order</Trans>}
+              onChange={(e, i, newValue) => {
+                // $FlowIgnore
```

Comment:
Add a quick explanation to ease the work for the future reader: // $FlowFixMe - newValue is blabla

Also FlowFixMe is the only officially supported name.